edweiss-firebase/functions/src/functions/forum/setLikePost.ts

```diff
@@ -0,0 +1,43 @@
+import { Timestamp } from 'firebase-admin/firestore';
+import { Forum } from 'model/forum';
+import { onSanitizedCall } from 'utils/firebase';
+import { CollectionOf, getDocumentAndRef } from 'utils/firestore';
+import { Predicate } from 'utils/sanitizer';
+import { OK } from 'utils/status';
+
+export const setLikePost = onSanitizedCall(Forum.Functions.setLikePost, {
+	courseId: Predicate.isNonEmptyString,
+	postId: Predicate.isNonEmptyString,
+	liked: Predicate.isBoolean,
+}, async (userId, args) => {
+	const [post, postRef] = await getDocumentAndRef(CollectionOf<Forum.Post>(`courses/${args.courseId}/forum`), args.postId);
+	const [like, likeRef] = await getDocumentAndRef(CollectionOf<Forum.Like>(`courses/${args.courseId}/forum/${args.postId}/likes`), userId);
```

Comment:
I don't get the impression here that you are testing whether the person has the course